### ISSUE TITLE
Support managed_challenge for custom_pages

### ISSUE THREAD

**Issue by fmalykh** [cloudflare/terraform-provider-cloudflare#1458]
### Current Terraform and Cloudflare provider version

Terraform v0.14.5
+ provider registry.terraform.io/cloudflare/cloudflare v3.9.1
+ provider registry.terraform.io/hashicorp/aws v4.0.0

### Description

Hi Team,

There is a new type of `custom_page` introduced - `managed_challenge`:

```
curl --silent -X GET "https://api.cloudflare.com/client/v4/accounts/xxx/custom_pages" \
     -H "X-Auth-Email: xxx" \
     -H "X-Auth-Key: xxx" \
     -H "Content-Type: application/json" | jq '.result[] | select(.id == "managed_challenge")'
{
  "id": "managed_challenge",
  "description": "Cloudflare Managed Challenge",
  "required_tokens": [
    "::CAPTCHA_BOX::"
  ],
  "preview_target": "block:managed-challenge",
  "created_on": "2022-02-15T14:23:03.826738Z",
  "modified_on": "2022-02-15T14:23:03.826738Z",
  "url": "xxx",
  "state": "customized"
}
```

Would be great to have it added to the `cloudflare_custom_pages` resource

### Use cases

Setting custom error page for `managed_challenge`

### Potential Terraform configuration

```hcl
resource "cloudflare_custom_pages" "managed_challenge" {
  zone_id = "d41d8cd98f00b204e9800998ecf8427e"
  type    = "managed_challenge"
  url     = "https://example.com/challenge.html"
  state   = "customized"
}
```

**Comment by jacobbednarz** [comment]
looks like this is still pending public documentation - https://api.cloudflare.com/#custom-pages-for-a-zone-properties

**Comment by georg-cabiri** [comment]
I need to set a custom page for **managed_challenge** but sadly the current terraform provider version (3.11.0) still does not validate the value **managed_challenge**

I get the following error still
```
Error: expected type to be one of [basic_challenge waf_challenge waf_block ratelimit_block country_challenge ip_block under_attack 500_errors 1000_errors always_o
nline], got managed_challenge   
```
I'm able to set the value through the dashboard and I can see **managed_challenge** used in API calls.
Any eta on when this will be supported as this is an urgent matter?
Thank you!

**Comment by jacobbednarz** [comment]
closed by #1478
